core/drivers/crypto/se050/adaptors/utils/scp_config.c

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ *
+ * This sequence follows the Global Platform Specification 2.2 - Amendment D
+ * for Secure Channel Protocol 03
+ *
+ */
+#include <assert.h>
+#include <bitstring.h>
+#include <crypto/crypto.h>
+#include <kernel/mutex.h>
+#include <kernel/refcount.h>
+#include <kernel/thread.h>
+#include <mm/mobj.h>
+#include <optee_rpc_cmd.h>
+#include <se050.h>
+#include <se050_default_keys.h>
+#include <se050_utils.h>
+#include <scp.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <tee_api_defines_extensions.h>
+#include <tee/tadb.h>
+#include <tee/tee_fs.h>
+#include <tee/tee_fs_rpc.h>
+#include <tee/tee_pobj.h>
+#include <tee/tee_svc_storage.h>
+#include <utee_defines.h>
+
+struct tee_scp03db_dir {
+	const struct tee_file_operations *ops;
+	struct tee_file_handle *fh;
+};
+
+static const char scp03db_obj_id[] = "scp03.db";
+static struct tee_pobj po = {
+	.obj_id_len = sizeof(scp03db_obj_id),
+	.obj_id = (void *)scp03db_obj_id,
+};
+
+static TEE_Result scp03db_delete_keys(void)__unused;
+static TEE_Result scp03db_delete_keys(void)
+{
+	struct tee_scp03db_dir *db = calloc(1, sizeof(struct tee_scp03db_dir));
+	TEE_Result res = TEE_SUCCESS;
+
+	assert(db);
+
+	db->ops = tee_svc_storage_file_ops(TEE_STORAGE_PRIVATE);
+	res = db->ops->open(&po, NULL, &db->fh);
+
+	/* nothing to delete */
+	if (res == TEE_ERROR_ITEM_NOT_FOUND) {
+		free(db);
+		return TEE_SUCCESS;
+	}
+
+	/* the filesystem might not be accessibe */
+	if (res != TEE_SUCCESS)
+		panic();
+
+	db->ops->close(&db->fh);
+	db->ops->remove(&po);
+	free(db);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result scp03db_write_keys(struct se050_scp_key *keys)
+{
+	struct tee_scp03db_dir *db = calloc(1, sizeof(struct tee_scp03db_dir));
+	TEE_Result res = TEE_SUCCESS;
+	size_t len = sizeof(*keys);
+
+	assert(db);
+
+	db->ops = tee_svc_storage_file_ops(TEE_STORAGE_PRIVATE);
+
+	res = db->ops->open(&po, NULL, &db->fh);
+	if (res != TEE_SUCCESS && res != TEE_ERROR_ITEM_NOT_FOUND) {
+		EMSG("scp03.db does not exist 0x%x", res);
+		free(db);
+		return TEE_ERROR_STORAGE_NOT_AVAILABLE;
+	}
+
+	IMSG("write scp03.db");
```

Comment:
consider moving many `IMSG()` trace to `DMSG()` in the P-R.

Reply:
ok. however for SCP03 it might be better if we let them be? the scp03 rotation is quite important as messing up the process bricks the se050 permanently.

Reply:
A trace is never a mandated information. I think this traces help you and SE050 developers to check the feature behavior, but are all those traces really generic informative traces?
In the end, I think it's up to you, as maintainer of this code to decide :)